### DISCUSSION TITLE
How can I use SSG with the app router?

### DISCUSSION THREAD

**Discussion by FleetAdmiralJakob** [vercel/next.js#61154] [App Router] [answered]
This is my ~offline page. But the page is dynamically rendered and not ssg. why?

```tsx
const OfflinePage = () => {
  return (
    <div className="flex w-full flex-col items-center">
      <h1 className="mt-10 text-4xl font-bold">Offline</h1>
      <p>
        It looks like you are offline. Please check your internet connection and
        try again.
      </p>
    </div>
  );
};

export default OfflinePage;
```

Route (app)                              Size     First Load JS
┌ ℇ /                                    3.02 kB         149 kB
├ λ /_not-found                          882 B          86.6 kB
├ λ /~offline                            137 B          85.9 kB <-------- NOT SSG
├ ℇ /api/trpc/[trpc]                     0 B                0 B
├ ℇ /post/[slug]                         3.02 kB         149 kB
├ ℇ /search                              3.02 kB         149 kB
├ ℇ /sign-in/[[...sign-in]]              212 B           106 kB
├ ℇ /sign-up/[[...sign-up]]              212 B           106 kB
└ ℇ /user/[slug]                         3.02 kB         149 kB
+ First Load JS shared by all            85.8 kB
  ├ chunks/30b509c0-046ad53b2a237fe0.js  53.4 kB
  ├ chunks/806-a48c935c341c5993.js       30.4 kB
  └ other shared chunks (total)          1.99 kB

**Comment by markoleavy** [comment]
I guess it's marked as lamda because you don't do any async request on server components, but the content of your page is still served as a cached html and rsc payload. Check the initial html on page requests, an try to add an async request and what happens.

**Comment by richardantao** [comment, marked as answer]
You can export a `dynamic` variable in your `~offline/page.(js|jsx|tsx)` to force SSG.

```
export const dynamic = 'force-static';
// 'auto' | 'force-dynamic' | 'error' | 'force-static'
```

Reference: [Docs](https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamic)
